pkg/services/ngalert/api/authorization.go

```diff
@@ -172,16 +172,17 @@ func (api *API) authorize(method, path string) web.Handler {
 	panic(fmt.Sprintf("no authorization handler for method [%s] of endpoint [%s]", method, path))
 }
 
-// GetDatasourceScopesFromAlertRule extracts data source scopes from an alert rule
-func getEvaluatorForAlertRule(rule *ngmodels.AlertRule) ac.Evaluator {
-	scopes := make([]ac.Evaluator, 0, len(rule.Data))
+// checkDatasourcePermissionsForRule checks that user has access to all data sources
+func checkDatasourcePermissionsForRule(rule *ngmodels.AlertRule, evaluator func(evaluator ac.Evaluator) bool) bool {
```

Comment:
nit: I'm not a fan of the word "check" as a verb for authorization. An alternative could be `verify` or `authorize` as checking seems a bit of an ambiguous word.